---
title: "Summarizing Multiomics Enrichment Analysis Results"
#author: "Zhiao Shi"
#date: "`r Sys.Date()`"
header-includes:  
  - \let\centering\relax
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

The main function of the package is `sumer()`. This function takes two parameters. The first parameter is the name of 
a file in json format that, among other things, specifies the location of input data files for each omics platform.  The second parameter
specifies the location of output files. Currently, enrichment results from up to 7 platforms can be summarized with `sumer()`.  

```r
sumer("/path/to/config.json", "/path/to/output_dir")
```

A sample configuration file is shown below:

```js
{
  "project":"My study",
  "top_num": 50,
  "data": [
    {
      "platform_name":"rna-seq",
      "platform_abbr":"rna",
      "gmt_file":"/data/rna_seq.gmt",
      "score_file":"/data/rna_seq_score.txt"
    },
    {
      "platform_name":"proteomics",
      "platform_abbr": "pro",
      "gmt_file":"/data/proteomics.gmt",
      "score_file":"/data/proteomics_score.txt"
    },
    {
      "platform_name":"phospho-proteomics",
      "platform_abbr": "phospho",
      "gmt_file":"/data/phospho_proteomics.gmt",
      "score_file":"/data/phospho_proteomics_score.txt"
    }
  ],
 "similarity": "Jaccard"
}
```

The keys for the configuration file are explained in the following table.

<style type="text/css">
.tg  {border-collapse:collapse;border-spacing:0;border-color:#ccc;}
.tg td{font-family:Arial, sans-serif;font-size:14px;padding:5px 5px;border-style:solid;border-width:0px;overflow:hidden;word-break:normal;border-color:#ccc;color:#333;background-color:#fff;}
.tg th{font-family:Arial, sans-serif;font-size:14px;font-weight:normal;padding:5px 5px;border-style:solid;border-width:0px;overflow:hidden;word-break:normal;border-color:#ccc;color:#333;background-color:#f0f0f0;}
.tg .tg-v47y{font-weight:bold;border-color:#000000;text-align:left}
.tg td{background-color:#f9f9f9;border-color:inherit;text-align:left}
<!-- .tg .tg-h6ay{background-color:#f9f9f9;border-color:inherit;text-align:left}
.tg .tg-xldj{border-color:inherit;text-align:left}
.tg .tg-btxf{background-color:#f9f9f9;border-color:inherit;text-align:left;vertical-align:top}
.tg .tg-0pky{border-color:inherit;text-align:left;vertical-align:top}
-->
</style>
<table class="tg" style="margin-top:10px; margin-left:0px;width:100%;">
  <tr>
    <th class="tg-v47y">name</th>
    <th class="tg-v47y">description</th>
  </tr>
  <tr>
    <td class="tg-h6ay">project</td>
    <td class="tg-h6ay">a brief description for your project</td>
  </tr>
  <tr>
    <td class="tg-xldj">top_num</td>
    <td class="tg-xldj">maximum number of top sets selected by set cover algorithm</td>
  </tr>
<tr>
    <td>similarity</td>
    <td>(optional) Similarity measure for gene sets, default is "Jaccard". The other supported
        measure is "Simpson".
        </td>
  </tr>
  <tr>
    <td class="tg-btxf">data               </td>
    <td class="tg-btxf">an array of objects that describe the data for each platform</td>
  </tr>
  <tr>
    <td class="tg-0pky"></td>
    <td class="tg-0pky"></td>
  </tr>
  <tr>
    <td class="tg-btxf">platform_name</td>
    <td class="tg-btxf">name of platform</td>
  </tr>
  <tr>
    <td class="tg-0pky">platform_abbr</td>
    <td class="tg-0pky">abbreviation of the platform name</td>
  </tr>
  <tr>
    <td class="tg-btxf"> gmt_file</td>
    <td class="tg-btxf">path to the gmt file</td>
  </tr>
  <tr>
    <td class="tg-0pky">score_file</td>
    <td class="tg-0pky">path to the score file</td>
  </tr>
</table>




The `gmt_file` specifies a tab delimited file that provides gene set information. Each row describes a gene set where the first 
column lists the name of gene set and the optional second column a brief description of the set. The rest of columns 
list the genes included in the set. This is the same format as described [here](https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMT:_Gene_Matrix_Transposed_file_format_.28.2A.gmt.29).

The `score_file` lists the measurement of significance for each gene set. There is one row for each gene set. The first column 
is the gene set name and second column is the score for the set. Typical examples of the score value are signed $-\log\{\mbox{p-value}\}$ or signed
$-\log\{\mbox{FDR}\}$. Columns are separated by tab.

The gene set names in `gmt_file` and `score_file` should match.

To run the sample data provided by the package, first make sure the working directory (`/path/to/your_work_dir`) and 
output directory (`/path/to/your_output_dir`) exist in your system.

```r
> setwd("/path/to/your_work_dir")
> file.copy(system.file("data", "sample.tgz", package="sumer"), ".")
> untar("sample.tgz")
> sumer("config.json", "/path/to/your_output_dir")
```

The final output files are in `your_output_dir`. You can open `index.html` to explore the summarized results. 
